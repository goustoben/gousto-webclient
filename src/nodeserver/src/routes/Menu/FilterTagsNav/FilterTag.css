

.filterTag {
    padding: 8px;
    border: 1px solid #D6D8DA;
    border-radius: 10px;
    box-shadow: inset 1px 0 0 0 #E9F0F7;
    text-align: center;
    margin-right: 10px;
    cursor: pointer;
}
