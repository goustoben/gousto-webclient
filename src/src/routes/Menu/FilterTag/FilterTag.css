@value Blackberry, Mackerel, Bluecheese, Blueberry, White from 'styles/colors.css';

.filterTag {
    border: 1px solid Mackerel;
    border-radius: 10px;
    text-align: center;
    margin-right: 10px;
    cursor: pointer;
    font-family: Avenir;
    font-size: 13px;
    font-weight: 900;
    line-height: 1.23;
    letter-spacing: normal;
    color: Bluecheese;
    &:hover {
        background-color: Mackerel;
    }
}

.selectedFilterTag {
    color: White;
    border-radius: 12px;
    border: solid 1px Blueberry;
    background-color: Bluecheese;
}

.tagIcon {
    padding-right: 5.5px;
}

.tagImageIcon {
    color: Blackberry;
}

.filterTagHeart {
	display: inline-flex;
	vertical-align: middle;
	justify-content: center;
	margin-left: 5px;
	width: 21px;
	height: 18px;
}
