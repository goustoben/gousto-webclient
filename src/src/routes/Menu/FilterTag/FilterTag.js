import PropTypes from 'prop-types'
import React from 'react'
import css from './FilterTag.css'

const proptypes = {
  text: PropTypes.string,
  value: PropTypes.string,
  selected: PropTypes.bool,
  filterDietaryAttributesChange: PropTypes.func,
}

const changeFilter = (filterDietaryAttributesChange, value) => {
  filterDietaryAttributesChange(value)
  if (window.pageYOffset > 250) {
    window.scrollTo(0,240)
  }
}

const FilterTag = ({selected, value, text, filterDietaryAttributesChange}) => (
  <div
    className={selected ? css.selectedFilterTag : css.filterTag}
    onClick={() => changeFilter(filterDietaryAttributesChange, value)}
  >
    {selected ? 
      <span className={css.tagIcon}>
        <span className={css.tagImageIcon} />
      </span>
      : null
    }
    {text}
  </div>
)

FilterTag.propTypes = proptypes

export default FilterTag
