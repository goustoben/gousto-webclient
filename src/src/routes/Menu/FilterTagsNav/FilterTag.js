import React, { PropTypes } from 'react'
import Icon from 'Icon'
import css from './FilterTag.css'

const FilterTag = ({ children }) => (
	<div className={css.filterTag}>
		<Icon name="fa-times" style={{ color: '#939596' }} />
		{children}
	</div>
)

FilterTag.propTypes = {
	children: PropTypes.node,
}

export default FilterTag
