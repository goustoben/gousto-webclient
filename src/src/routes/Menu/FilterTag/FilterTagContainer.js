import { connect } from 'react-redux'

import FilterTag from './FilterTag'
import filterActions from 'actions/filters'

const FilterTagContainer = connect(() => ({}), {
	collectionFilterChange: filterActions.filterCollectionChange,
	filterCurrentDietTypesChange: filterActions.filterCurrentDietTypesChange,
	filterDietaryAttributesChange: filterActions.filterDietaryAttributesChange,
	filterCurrentTotalTimeChange: filterActions.filterCurrentTotalTimeChange,
})(FilterTag)

export default FilterTagContainer
