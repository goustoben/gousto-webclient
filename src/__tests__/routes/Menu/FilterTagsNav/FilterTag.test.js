import React from 'react';
import renderer from 'react-test-renderer';
import FilterTag from 'routes/Menu/FilterTagsNav/FilterTag'

describe("FilterTag", () => {

  describe("rendering", () => {

    test("Will render FilterTag", () => {
      const tree = renderer
        .create(<FilterTag>Facebook</FilterTag>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    })
  })
})
