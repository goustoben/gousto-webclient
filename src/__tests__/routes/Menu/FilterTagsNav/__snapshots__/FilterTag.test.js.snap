// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTag rendering Will render FilterTag 1`] = `
<div
  className="filterTag"
>
  <span
    className="fa-times"
    style={
      Object {
        "color": "#939596",
        "fontSize": "inherit",
      }
    }
  />
  Facebook
</div>
`;
